sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/FlatteningDeserializer.java

```diff
@@ -188,30 +188,34 @@ private void handleFlatteningForField(AnnotatedField annotatedField, JsonNode js
                 String[] jsonNodeKeys = Arrays.stream(SPLIT_KEY_PATTERN.split(jsonPropValue))
                     .map(FlatteningDeserializer::unescapeEscapedDots)
                     .toArray(String[]::new);
-
+                int depth = 0;
                 // Keep track of the JsonNodes which lead to the flattened property being deserialized.
                 //
                 // This will be used later to clean up the parent JsonNode so that there aren't additional dangling
                 // JSON once the flattened value is uplifted. If this isn't done it can result in models with
                 // @JsonAnySetter containing errant values.
                 List<JsonNode> nodePath = new ArrayList<>();
                 nodePath.add(jsonNode);
+                depth++;
                 JsonNode nodeToAdd = jsonNode;
                 for (String jsonNodeKey : jsonNodeKeys) {
                     nodeToAdd = nodeToAdd.get(jsonNodeKey);
+                    depth++;
```

Comment:
I believe this increment should happen after the null check

Reply:
I think it should always be incremented and be irrespective of whether the value of the node is found or not.